src/test/SampleCustomLoader/sampleCustLoader.cpp

```diff
@@ -0,0 +1,320 @@
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+#include <chrono>
+#include <fstream>
+#include <future>
+#include <iostream>
+#include <map>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <assert.h>
+#include <rapidjson/document.h>
+
+#include "../../customloaderinterface.hpp"
+
+using namespace std;
+using namespace rapidjson;
+using namespace ovms;
+
+#define PATH_SIZE 10
+#define RSIZE_MAX_STR 4096
+
+#define CUSTLOADER_OK 1
+#define CUSTLOADER_ERROR 40
+
+typedef pair<string, int> model_id_t;
+
+class custSampleLoader : public CustomLoaderInterface {
+private:
+    vector<model_id_t> models_blacklist;
+    vector<model_id_t> models_loaded;
+
+protected:
+    int extract_input_params(char* basePath, int version, char* loaderOptions, char** binFile,
+        char** xmlFile);
+    int load_files(char* binFile, char* xmlFile, char** xmlBuffer, char** binBuffer, int* xmlLen,
+        int* binLen);
+
+    std::promise<void> exitSignal;
+    std::future<void> futureObj;
+    int watchIntervalSec = 0;
+    bool watcherStarted = false;
+    std::thread watcher_thread;
+
+public:
+    custSampleLoader();
+    ~custSampleLoader();
+    int loaderInit(char* loader_path);
+    void loaderDeInit();
+    int unloadModel(const char* modelName, int version);
+    int loadModel(const char* modelName, const char* basePath, const int version,
+        const char* loaderOptions, char** xmlBuffer, int* xmlLen, char** binBuffer,
+        int* binLen);
+    bool getModelBlacklistStatus(const char* modelName, int version);
+
+    void threadFunction(std::future<void> futureObj);
+    void startWatcher(int intervalSec);
+    void watcherJoin();
+};
+
+extern "C" CustomLoaderInterface* create() {

```

Comment:
symbol named "create" just asks for symbol name conflict. Please use some prefix in the name in all those.